content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -0,0 +1,176 @@
+---
+title: 'Relation mode'
+metaTitle: 'Relation mode'
+metaDescription: 'Relation mode'
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+In relational databases, relations between data are usually implemented with foreign keys. A relation consists of a target row in a table, and a foreign key field that points to that row.
+
+Foreign keys:
+
+- ensure referential integrity (Can this foreign key field hold this value?)
+- allow you to set referential actions that define how changes to these fields are handled (What happens if the ID of this field is changed?).
+
+In Prisma, relations are defined using the [`@relation`](/reference/api-reference/prisma-schema-reference#relation) attribute, which then under the hood is translated to foreign keys and the SQL to define referential actions (which are defined with the `onUpdate` and `onUpdate` arguments of the attribute).
+
+!!explain example
+
+<!--
+Prisma allows you to define relations between records using the [`@relation`](/reference/api-reference/prisma-schema-reference#relation) attribute. The `onUpdate` and `onDelete` arguments of this attribute can be used to define which referential action take place on updating or deleting the related record.
+-->
+
+<!-- TODO Tab that shows SQL -->
+
+```prisma file=schema.prisma highlight=4;normal
+model Post {
+  id       Int    @id @default(autoincrement())
+  title    String
+  author   User   @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+Some databases, such as MongoDB or [PlanetScale](/guides/database/using-prisma-with-planetscale#differences-to-consider)),
+do not support foreign keys. For this situation, Prisma offers an alternative _relation mode_ where most properties of relations in relational databases are emulated.
+
+When using Prisma Client the behavior of queries is identical or similar, but all referential actions and most referential integrity is actually handled by the Prisma engine under the hood.
+
+<Admonition type="warning">
+  There are significant performance implications to emulating referential
+  integrity in the Prisma client. In cases where the underlying database can
+  handle referential integrity with foreign keys, it is usually the preferred
+  choice.
+</Admonition>
+
+</TopBlock>
+
+## How to set the relation mode in your Prisma schema
+
+To set the relation mode, you need to enable the `referentialIntegrity` preview feature. Add the `referentialIntegrity` feature flag to the `previewFeatures` list in the [`generator`](/reference/api-reference/prisma-schema-reference#generator) block of your Prisma schema file, and add the `relationMode` field to the [`datasource`](/reference/api-reference/prisma-schema-reference#datasource) block:
```

Comment:
This could be rewritten in a way that it first only talks about `relationMode`. 
And then "As long as this is in preview, you also need to enable the preview feature by ...".

Reply:
Yes, I prefer this - have rearranged.